### PR TITLE
nonroot CentOS Docker image

### PR DESCRIPTION
**Pull request by JohnnySheffield**
@niklasad1 @General-Beck Continuation of #9231, but with a much cleaner commit history.

### CODE REVIEW

File: docker/centos/build.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+FROM centos:latest
```
Comment (debris) [inline docker/centos/build.Dockerfile]: why is this file called `build.Dockerfile`? Is it a common convention? (I'm not a docker expert)

File: docker/centos/build.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+WORKDIR /build
+
+ADD . /build/parity-ethereum
+
+RUN yum -y update && \
+    yum install -y systemd-devel git make gcc-c++ gcc file binutils && \
+    curl -L "https://cmake.org/files/v3.12/cmake-3.12.0-Linux-x86_64.tar.gz" -o cmake.tar.gz && \
```
Comment (debris) [inline docker/centos/build.Dockerfile]: why not `yum install cmake` ?

Comment (JohnnySheffield) [reply docker/centos/build.Dockerfile]: yum install cmake installs cmake 2.x, but then parity build fails as it needs 3.x, so i decided to install cmake directly. Ideally i would like to get it with yum, but had no luck

File: docker/centos/build.Dockerfile (hunk 1 above)
Comment (JohnnySheffield) [reply docker/centos/build.Dockerfile]: I'll rename it to Dockerfile.build, short google points that this is some kind of convention

**Pull request merged**